### PR TITLE
Add tests for observabilitytest, change api to return error.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: observability/observabilitytest/observabilitytest.go [hunk 1]
```diff
@@ -43,27 +43,27 @@ func (cme *nopMetricsExporter) ExportView(vd *view.Data) {}
 
 // SetupRecordedMetricsTest does setup the testing environment to check the metrics recorded by receivers, producers or exporters.
 // The returned function should be deferred "defer SetupRecordedMetricsTest(t)()".
```
Comment (songy23) [inline observability/observabilitytest/observabilitytest.go]: nit: this comment should also be updated.

File: observability/observabilitytest/observabilitytest_test.go [hunk 2]
```diff
@@ -12,6 +12,134 @@
...
+func TestNoSetupCalled(t *testing.T) {
+	receiverCtx := observability.ContextWithReceiverName(context.Background(), receiverName)
+	observability.RecordTraceReceiverMetrics(receiverCtx, 17, 13)
+	// Failed to check because views are not registered.
+	if err := observabilitytest.CheckValueViewReceiverReceivedSpans(receiverName, 17); err == nil {
+		t.Fatalf("When check recorded values: want not-nil got nil")
+	}
+}
+
+func TestCheckRecordedMetricsForTraceExporter_WithNoMetricsRecorded(t *testing.T) {
```
Comment (songy23) [inline observability/observabilitytest/observabilitytest_test.go]: I assume there will be another PR on tests on MetricsExporter?

File: exporter/exportertest/nop_exporter_test.go [hunk 3]
```diff
@@ -15,39 +15,64 @@ package exportertest
...
+	}
+	if got := nte.ConsumeTraceData(context.Background(), td); got != want {
+		t.Fatalf("Want %v Got %v", want, got)
+	}
+	if "nop_trace" != nte.TraceExportFormat() {
+		t.Fatalf("Wanted nop_trace got %s", nte.TraceExportFormat())
+	}
+}
+
+func TestNoopMetricsExporter_NoErrors(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply exporter/exportertest/nop_exporter_test.go]: Done.

File: observability/observabilitytest/observabilitytest.go [hunk 4]
```diff
@@ -43,27 +43,27 @@ func (cme *nopMetricsExporter) ExportView(vd *view.Data) {}
...
 	// Register a nop metrics exporter for the OC library.
 	nmp := new(nopMetricsExporter)
 	view.RegisterExporter(nmp)
 
 	// Now for the stats exporter
-	if err := view.Register(observability.AllViews...); err != nil {
-		t.Fatalf("Failed to register all views: %v", err)
-	}
+	view.Register(observability.AllViews...)
+	fmt.Printf("Setup Executed.\n")
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply observability/observabilitytest/observabilitytest.go]: Done.

File: observability/observabilitytest/observabilitytest.go (hunk 1 above)
Comment (bogdandrutu) [reply observability/observabilitytest/observabilitytest.go]: Done.

File: observability/observabilitytest/observabilitytest_test.go (hunk 2 above)
Comment (bogdandrutu) [reply observability/observabilitytest/observabilitytest_test.go]: Yes.
